synthesizer/process/src/stack/mod.rs

```diff
@@ -261,6 +262,16 @@ impl<N: Network> StackProgram<N> for Stack<N> {
         }
     }
 
+    /// Returns a reference to the function with the given function name.
+    #[inline]
+    fn get_function_ref(&self, function_name: &Identifier<N>) -> Result<&Function<N>> {
+        // Ensure the function exists.
+        match self.program.contains_function(function_name) {
+            true => self.program.get_function_ref(function_name),
+            false => bail!("Function '{function_name}' does not exist in program '{}'.", self.program.id()),
+        }
```

Comment:
This `contains_function` call is extraneous. If the function does not exist in the program, `get_function_ref` will fail in the first line: `let function = self.functions.get(name).ok_or_else(|| anyhow!("Function '{name}' is not defined."))?;
`


```suggestion
        self.program.get_function_ref(function_name)
```